### ISSUE TITLE
Stubber should inherit from contextlib.AbstractContextManager

### ISSUE THREAD

**Issue by tibbe** [boto/botocore#2873]
### Describe the feature

Stubber is a context manager (e.g. it has all the required methods) but doesn't inherit from `contextlib.AbstractContextManager`. This means that it can't be used with e.g. `contextlib.ExitStack`.

### Use Case

Using `Stubber` with the Python context manager ecosystem.

### Proposed Solution

Probably a one line change to add the base class.

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### SDK version used

1.29.73

### Environment details (OS name and version, etc.)

macOS

**Comment by tibbe** [comment]
My use case is to use the `Stubber` as a context manager with the Python standard library, in particular https://docs.python.org/3/library/contextlib.html. Context managers in Python should inherit from `contextlib.AbstractContextManager`. If they don't standard type checkers, like mypy, will not be able to type check code using `Stubber` correctly e.g. when using it with `contextlib.ExitStack`.

**Comment by tibbe** [comment]
Having looked into this more `Stubber` shouldn't have to inherit from `AbstractContextManager`. It just needs a correct type annotation on `__exit__`. Currently it is:

```py
def __exit__(self, exception_type, exception_value, traceback): ...
```

but it needs to be:

```py
def __exit__(
    self, exc_type: Optional[Type[BaseException]], exc: Optional[BaseException], traceback: Optional[TracebackType]
) -> Optional[bool]: ...
```

I've recently fixed the same problem in another package so we know this is the right fix.

**Comment by tibbe** [comment]
It's not entirely clear to me if this type annotation should live in this project or mypy_boto3_builder. I filed https://github.com/youtype/mypy_boto3_builder/issues/189 against mypy_boto3_builder.

**Comment by vemel** [comment]
Thank you for the report!

Fixed in `botocore-stubs 1.29.93.post1`. Please test this new release and let me know if it works as expected.
